### ISSUE TITLE
[Warrior] T10 4P Bonus

### ISSUE THREAD

**Issue by Jildor** [tobmaps/TrinityCore#25]
It's bug

I think this fix solves:

http://www.trinitycore.org/t/trinitycore/ticket/449#comment:7

**Comment by tobmaps** [comment]
this patch isn't working properly. I does remove all charges at using Slam and Execute and Slam proc doesn't has 2 charges

**Comment by Jildor** [comment]
Oks.

So, any solution?

thx

**Comment by tobmaps** [comment]
it's not so easy to fix charge consuming for this... I'm no planning to fix it for a while, i more prefer to fix talents and player spells ^^

**Comment by Jildor** [comment]
mmmm

Are you applied the sql to patch?
http://www.trinitycore.org/t/trinitycore/attachment/ticket/449/patch.sql
I think works.

**Comment by tobmaps** [comment]
yes i've applied this, but anyway this SQL will not fix bugs that i meant before

**Comment by Jildor** [comment]
I count as 2 charges and I removed one at a time....I don't know :S

**Comment by kbinside** [comment]
important bug, waiting for fix.

**Comment by Jildor** [comment]
possible fix?
https://github.com/SignFinder/FaceCore/commit/085c8d8d17bfd92595db8a47ec3d93e622be9ef8

**Comment by kbinside** [comment]
great, add this.

**Comment by tobmaps** [comment]
this fix is a one big hack, correct solution can be done when QAston will done his new proc system

**Comment by kbinside** [comment]
its about time, to make this bonus working?

**Comment by tobmaps** [comment]
better to wait for implementation of new proc system by QAston and only then try to fix it. For now you can easily use any hack that you want, i just will not review it

**Comment by kbinside** [comment]
okey, keep waiting.

maybe you can check: https://github.com/TrinityCore/TrinityCore/issues/2014
thanks
